corehq/apps/app_manager/helpers/validators.py

```diff
@@ -47,7 +47,8 @@ def validate_app(self):
             self._validate_fixtures()
             self._validate_intents()
             self._validate_practice_users()
-            self.app.create_all_files()
+            if not errors:
```

Comment:
I think `errors` needs to be made into an instance variable for this to work as I think is intended: that if `ApplicationValidator` (the subclass) comes up with errors, then this method isn't called.

Otherwise I think the only possible errors that `errors` could contain here are from `_check_password_charset`.

This might be out of scope for this refactor, but would it be wise to move these into their own try/except blocks? That way all errors can be caught in a single pass, rather than here where after the first error is caught, the function execution skips over the next validators and the user has to fix one at a time. Or are these all related to each other, where if the first one fails, the next won't be able to run?

Reply:
> I think errors needs to be made into an instance variable for this to work

I don't understand this. My intention is that if the subclass finds errors, it does still call this method, it just doesn't attempt to `create_all_files`. Any errors found here are then added to what the subclass already found.

I took a stab at reworking the exception blocks in 574cb6513ad - it isn't ideal that they throw exceptions rather than returning errors and allowing validation to proceed. I'm reasonably confident that the `_validate_foo` helpers don't interact with each other.